builtin/providers/aws/resource_aws_elb.go

```diff
@@ -211,10 +213,18 @@ func resourceAwsElbCreate(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
+	var elbName string
+	if v, ok := d.GetOk("name"); ok {
+		elbName = v.(string)
+	} else {
+		elbName = resource.PrefixedUniqueId("tf-lb-")
+		d.Set("name", elbName)
```

Comment:
This is here because of update function being called at the end, it feels a bit dirty, but I can't think of better/cleaner solution right now.